src/types/schema.ts

```diff
@@ -1,65 +1,105 @@
 import { UUID } from 'crypto';
 
-export interface CaseListing {
-  id: UUID;
-  title: string;
-  summary: string;
-  country: string;
-  client_location: string;
-  legal_server_id: number;
-  hours_per_month: number;
-  num_months: number;
-  languages: string[];
-  is_remote: boolean;
-  in_court: boolean;
-  needs_attorney: boolean;
-  needs_interpreter: boolean;
-  upcoming_date: string;
-  relief_sought: string;
-  active: boolean;
+// table field types
+export type ImmigrationLawExperienceEnum = 'LOW' | 'MEDIUM' | 'HIGH';
+
+export type AdjudicatingAgencyEnum =
+  | '9TH_CIRCUIT'
+  | 'BIA'
+  | 'IMMIGRATION_COURT'
+  | 'ICE'
+  | 'USCIS';
+
+export type RoleEnum =
+  | 'ATTORNEY'
+  | 'INTERPRETER'
+  | 'LEGAL_FELLOW'
+  | 'TRANSLATOR';
+
+export type ListingTypeEnum =
+  | 'CASE'
+  | 'LIMITED_ASSISTANCE'
+  | 'TRANSLATION_REQUEST';
+
+// join table rows
+export interface ProfileLanguage {
+  user_id: UUID;
+  iso_code: string;
+  can_read: boolean;
+  can_write: boolean;
 }
 
-export interface LimitedAssistance {
-  id: UUID;
-  summary: string;
-  languages: string[];
-  country: string;
-  experience_level: string;
-  deadline: string;
-  interest_ids: UUID[];
+export interface ProfileRole {
+  user_id: UUID;
+  role: RoleEnum;
 }
 
-export interface TranslationRequest {
+export interface CaseLanguage {
+  listing_id: UUID;
+  iso_code: string;
+}
+
+export interface CaseRelief {
+  listing_id: UUID;
+  relief_code: string;
+}
+
+// sql table rows
+export interface CaseListing {
   id: UUID;
-  summary: string;
-  languages: string[];
-  interest_ids: UUID[];
+  title?: string;
+  summary?: string;
+  country?: string;
+  client_location?: string;
+  legal_server_id: number;
+  hours_per_month?: number;
+  num_months?: number;
+  is_remote?: boolean;
+  needs_attorney?: boolean;
+  needs_interpreter?: boolean;
+  upcoming_date?: string;
+  experience_needed: ImmigrationLawExperienceEnum;
+  adjudicating_agency: AdjudicatingAgencyEnum;
 }
 
 export interface Profile {
   user_id: UUID;
   first_name: string;
   last_name: string;
   location: string;
-  roles: string[];
   hours_per_month: number;
-  immigration_law_experience: string;
-  languages: string[];
-  bar_number: string;
+  immigration_law_experience: ImmigrationLawExperienceEnum;
+  bar_number?: string;
   start_date: string;
-  availability_description: string;
+  availability_description?: string;
   eoir_registered: boolean;
-  interest_ids: UUID[];
 }
 
 export interface Interest {
-  id: UUID;
   listing_id: UUID;
   user_id: UUID;
-  listing_type: number;
+  listing_type: ListingTypeEnum;
   form_response: {
-    interestType: string;
+    rolesInterested: RoleEnum[];
     interestReason: string;
     start_date: string;
```

Comment:
is it not possible to make `start_date` of type `timestamptz`?

Reply:
I'll make it a js Date, thanks for flagging that!